packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -40,20 +40,22 @@ class MediaQueryData {
     this.size: Size.zero,
     this.devicePixelRatio: 1.0,
     this.textScaleFactor: 1.0,
-    this.padding: EdgeInsets.zero
-  });
+    this.padding: EdgeInsets.zero,
+    TargetPlatform platform
+  }) : _platform = platform;
 
   /// Creates data for a media query based on the given window.
   ///
   /// If you use this, you should ensure that you also register for
   /// notifications so that you can update your [MediaQueryData] when the
   /// window's metrics change. For example, see
   /// [WidgetsBindingObserver.didChangeMetrics] or [Window.onMetricsChanged].
-  MediaQueryData.fromWindow(ui.Window window)
+  MediaQueryData.fromWindow(ui.Window window, {TargetPlatform platform})
     : size = window.physicalSize / window.devicePixelRatio,
       devicePixelRatio = window.devicePixelRatio,
       textScaleFactor = 1.0, // TODO(abarth): Read this value from window.
-      padding = new EdgeInsets.fromWindowPadding(window.padding, window.devicePixelRatio);
+      padding = new EdgeInsets.fromWindowPadding(window.padding, window.devicePixelRatio),
+      _platform = platform ?? defaultTargetPlatform;
```

Comment:
Setting _platform here will mean that when you toggle the platform (e.g. by hitting "o" in "flutter run") the platform won't properly propagate all the way through unless we force a rebuild.